### ISSUE TITLE
Implement caching

### ISSUE THREAD

**Issue by neefrehman** [kemiljk/better-logos-figma#23]
We should cache the results of the fetch in `localStorage`, and read from that if it exists, instead of performing another fetch on pageload. We should also have a timestamp to invalidate the cache after say, one day?

**Comment by kemiljk** [comment]
Yep agreed. One good thing with Cosmic here is that api requests are cached via CDN and served from that cache within a certain TTL window. But I think localStorage would be good for us and also decouples the architecture from the plugin itself.

**Comment by kemiljk** [comment]
Basically I think the current implementation might be okay for initial release but localStorage will be super useful down the line unless it ends up easy enough to implement or valuable enough.

**Comment by neefrehman** [comment]
Following up on the other point here @kemiljk:

>Just a thought, could we actually access the figma context via the pluginMessage API? So have a request in ui.html that triggers storage or gets from storage? ui.ts will accept the figma context so could that be a route around using the standard window controls? Depends if it's necessary really, but the getAsync() method is async which is handy.

We could pass the stringified JSON to figma using the pluginMessage API, but I think it's still best to use localStorage. It's the UI that will need to cache the JSON, so having to retrieve it through figma's storage (via a pluginMessage) is an extra layer that we don't really need.

I'll do a quick localStorage implementation today or tomorrow.

**Comment by kemiljk** [comment]
Haha 😂

**Comment by neefrehman** [comment]
Ok just tried and it looks like our response is too large for `clientStorage` to handle:

```shell
unhandled promise rejection: Throw: Failed to set client storage key "cachedLogoData": Client storage quota exceeded for this plugin ID. Please use less client storage.
```

Maybe caching isn't possible with our current implementation. I guess long-term if we move to fetch-on-type we also couldn't cache properly anyway. Maybe we should park this one?

We could also try compressing the JSON and decompressing it at the other end? But I just fear that will break too when we add more logos...
